### PR TITLE
Add validation to config changes

### LINKED ISSUE #31: Install Promtool
We should install the promtool so that configuration can be checked before reloading the service.

```
./promtool -help
usage: promtool <command> [<args>]

Available commands:
  check-config    validate configuration files for correctness
  check-rules     validate rule files for correctness
  help            prints this help text
  version         print the version of this binary
```

### PR DESCRIPTION
**Pull request by costela**
This PR adds a symlink to promtool in the same way a symlink to prometheus is currently added to /usr/local/bin.
Than it uses File's `validate_cmd` to call promtool and avoid replacing a configuration file with a broken version, both for the central config and for the alert rules.

Finally, also fix the first error thrown by the config check: we create the alerts file even if no alerts were provided. The empty file won't trigger a syntax error and also doesn't bother prometheus itself.

Fixes #31

### CODE REVIEW

**Comment by tuxmea** [comment]
@costela Many thanks for the PR. Please check the travis error:
Evaluation Error: Unknown variable: 'prometheus::alerts::location'. at manifests/config.pp:127:29

**Comment by costela** [comment]
@tuxmea yeah, I'm on it. Had a last minute change of heart on the last patch, and it didn't go very smoothly :wink:

**Comment by costela** [comment]
Please note there's a small breaking change in the last patch: the `purge` parameter to `prometheus::config` was renamed and moved to `prometheus::install`, together with the creation of the `/etc/prometheus` directory. This was necessary to add the missing dependency between the config file and the alerts file without creating a dependency cycle.
Since I don't expect direct use of `prometheus::config` to be common (or supported), I assumed it was an acceptable change.

**Comment by costela** [comment]
@tuxmea did you maybe have a minute to check the PR?
Don't mean to be a nuisance, but I have a second, bigger PR in the pipeline for #126 that I'm too lazy to rebase on master :wink:

File: manifests/install.pp [hunk 1]
```diff
@@ -3,8 +3,9 @@
 # Currently only the install from url is implemented, when Prometheus will deliver packages for some Linux distros I will
 # implement the package install method as well
 # The package method needs specific yum or apt repo settings which are not made yet by the module
-class prometheus::install
-{
+class prometheus::install (
+  Boolean $purge_config_dir = true,
```
Comment (bastelfreak) [inline manifests/install.pp:7]: Datatypes \o/

**Review by tuxmea — COMMENTED** [review]
Is there a specific reason why config_dir was moved from config.pp to install.pp?

File: manifests/install.pp [hunk 2]
```diff
@@ -78,4 +82,11 @@
       system => true,
     })
   }
+  file { $prometheus::config_dir:
```
[the remark this answers is not in the document]
Comment (costela) [reply manifests/install.pp:85]: as I mentioned in a comment above: this was necessary to add the missing dependency between the config file and the alerts file without creating a dependency cycle. Now both depend on the directory, which gets created first during install.
The config file needs the alerts file, because the newly introduced validation with `promtool` chokes if it can't find the named alerts file.
That's also the reason why I removed the conditional creation of the alerts file and create it always (even if empty). Otherwise the dependency would have to be conditional on the alert file's creation and the `prometheus.yaml.erb` template would also have to be changed to not always reference a possibly non-existent alerts file.
In summary: it just felt like the path of least resistance :wink:

[the remark this answers is not in the document]
Comment (tuxmea) [reply manifests/install.pp:85]: ACK +1
